skyportal/facility_apis/swift.py

```diff
@@ -150,6 +244,131 @@ def _build_payload(self, request):
         return myReq
 
 
+class UVOTXRTBATDataRequest:
+
+    """A JSON structure for Swift UVOT/XRT/BAT Data requests."""
+
+    def __init__(self, request):
+        """Initialize UVOT/XRT/BAT data request.
+
+        Parameters
+        ----------
+        request : skyportal.models.FollowupRequest
+            The request to add to the queue and the SkyPortal database.
+
+        """
+
+        self.requestgroup = self._build_payload(request)
+
+    def _build_payload(self, request):
+        """Payload json for Swift UVOT/XRT/BAT data requests.
+
+        Parameters
+        ----------
+        request : skyportal.models.FollowupRequest
+            The request to add to the queue and the SkyPortal database.
+
+        Returns
+        -------
+        payload : swifttools.swift_too.ObsQuery
+            payload for requests.
+        """
+
+        from swifttools.swift_too import ObsQuery
+
+        oq = ObsQuery(
+            ra=request.obj.ra,
+            dec=request.obj.dec,
+            radius=5.0 / 60,
+            begin=request.payload["start_date"],
+            end=request.payload["end_date"],
+        )
+
+        return oq
+
+
+def download_observations(request_id, oq):
+    """Fetch data from the Swift API.
+    request_id : int
+        ID for request
+    oq : swifttools.swift_too.ObsQuery
+        Swift observation query
+    """
+
+    from ..models import (
+        Comment,
+        DBSession,
+        FollowupRequest,
+        Group,
+    )
+    from swifttools.swift_too import Data
+
+    Session = scoped_session(sessionmaker(bind=DBSession.session_factory.kw["bind"]))
+    session = Session()
```

Comment:
Can you remind me again why we are not using the regular handler and all the tools that go with it?

Reply:
It is because of all the external API calls in the facility handlers.